src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Azure/Functions/IFunctionInvoker.cs

```diff
@@ -0,0 +1,14 @@
+// <copyright file="IFunctionInvoker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Azure.Functions
+{
+    /// <summary>
+    /// For duck typing
+    /// </summary>
+    public interface IFunctionInvoker
+    {
+    }
```

Comment:
Do we need this interface if we're not accessing anything? If not, then should probably remove to avoid unnecessary ducktyping overhead?

Reply:
This is the actual instance that the method is invoked on. I hadn't found any examples of `object` being used, but perhaps I missed them. @tonyredondo Do you have thoughts on this?

Reply:
My not-Tony suggestion would be to remove this interface and just have the `TTarget` generic type not implement any interfaces. It will just implement `object` at that point

Reply:
☝️ yeah, if there are no requirements to call methods on the instance being invoked, you can just remove the constraint. This is pretty common, e.g there's no constraint on `TTarget` here:

https://github.com/DataDog/dd-trace-dotnet/blob/fff09c81d2cc10c74655b3b7a5c06b8c4d99da29/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaProduceSyncIntegration.cs#L41-L42